recipe/meta.yaml

```diff
@@ -13,32 +15,79 @@ source:
   sha256: 23793c8645480148e9559df96b386b780f92194c80120acce79fcdaae0d81f45
 
 build:
-  number: 0
-  script: python build/build.py {{ args }}                                          # [unix]
-  script: python build/build.py --make-tool=mingw32-make {{ args }}                 # [win]
+  number: 1
   script_env:
     - OS    # [win]
-  features:
-    - vc9   # [win and py27]
-    - vc14  # [win and py>=35]
 
 requirements:
   build:
     - toolchain
     - python
     - swig
-    - m2w64-make  # [win]
+    - {{ compiler('cxx') }}
+    - m2w64-make                                         # [win]
 
 test:
   requires:
-    - python
+    - python *               # any python version is ok for sake of testing of shared libraries
   commands:
-    - test -f ${PREFIX}/include/tbb/tbb.h  # [unix]
-    - python -c "import ctypes; ctypes.cdll[r'${PREFIX}/lib/libtbb.so.2']"  # [linux]
-    - python -c "import ctypes; ctypes.cdll[r'${PREFIX}/lib/libtbb${SHLIB_EXT}']"  # [unix and not linux]
-    - python -c "import ctypes; ctypes.cdll[r'%PREFIX%\Library\bin\tbb.dll']"  # [win]
-    - if not exist %PREFIX%\\Library\\include\\tbb\\tbb.h exit 1  # [win]
-    - if not exist %PREFIX%\\Library\\lib\\tbb.lib exit 1  # [win]
+    - python -c "import ctypes; assert {{ vinterface }} == ctypes.cdll[r'libtbb.so.2']       ['TBB_runtime_interface_version']()"  # [linux]
+    - python -c "import ctypes; assert {{ vinterface }} == ctypes.cdll[r'libtbb${SHLIB_EXT}']['TBB_runtime_interface_version']()"  # [unix and not linux]
+    - python -c "import ctypes; assert {{ vinterface }} == ctypes.cdll[r'tbb.dll']           ['TBB_runtime_interface_version']()"  # [win]
+
+outputs:
+  - name: tbb
+    build:
+      script: {{ build }} --install-libs
+      binary_relocation: False
+
+  - name: tbb-devel
+    build:
+      script: {{ build }} --no-rebuild --install-devel --install-docs
+      binary_relocation: False
+    requirements:
+      host:
+        - {{ compiler('cxx') }}
+        - m2w64-make                                     # [win]
```

Comment:
These should both be in `build` instead of `host` as they are build tools that don't need to live in the install prefix. IIRC we still want an empty `host` section. @msarahan would know for sure though.

Reply:
done, though, it looks like I have to repeat myself for the sub-packages. Wouldn't OOP-like relationships between global and sub-package-specific sections be better approach in meta.yaml?